docs/reference/docs/reindex.asciidoc

```diff
@@ -0,0 +1,461 @@
+[[docs-reindex]]
+==== Reindex API
+
+`_reindex`'s most basic form just copies documents from one index to another.
```

Comment:
I have a bunch of other [PRs](https://github.com/elastic/elasticsearch/issues/17031) to backport for reindex that will help this a lot. I don't expect those PRs to have their own backport PRs because I expect those backports to be trivial.

#16618, I think, will help even more. But right now the branches are so out of sync these docs are just here to make the branches *closer*.